web/packages/teleport/src/Discover/Shared/SecurityGroupPicker/SecurityGroupRulesDialog.tsx

```diff
@@ -124,3 +123,17 @@ const StyledTable = styled(Table)`
   box-shadow: ${props => props.theme.boxShadow[0]};
   overflow: hidden;
 ` as typeof Table;
+
+function getPortRange(
+  ipProtocol: string,
+  fromPort: string,
+  toPort: string
+): string {
+  if (ipProtocol === 'all') {
+    // fromPort and toPort are irrelevant and AWS currently returns 0 for both
+    // in this case.
+    return 'all';
+  }
+  // If they are the same, only show one number.
+  return fromPort === toPort ? fromPort : `${fromPort} - ${toPort}`;
+}
```

Comment:
oh, and I added this because I realized when a rule has "all" protocol allowed, AWS returns fromPort=toPort=0 and our dialog box was showing the port range as `0` as a result.
This function makes it just show in that case `all` for the port range instead.